robotoff/workers/tasks/update_recycle.py

```diff
@@ -0,0 +1,206 @@
+    :param username:
+    :param password:
+    :return:
+    """
+    field = "packaging_{}".format(lang)
+
+    data = get_json_for_image(product.get("code"), image)
+    image_text = data.get("responses", [])
+    if image_text:
+        image_text = image_text[0].get("fullTextAnnotation", {}).get("text", "")
+
+        if any([1 if i in image_text.lower() else 0 for i in recycling_triggers]):
+            ru = None
+            if field in pack_images.keys():
+                ru = unselect_image(ean, field, username, password)
+
+            r = reselect_image(ean, field, image, username, password)
+
+            if r:
+                if ru:
+                    logger.info(
+                        "Recycle image(changed): ",
+                        ean,
+                        field,
+                        pack_images[field].get("imgid"),
+                        "->",
+                        image,
+                    )
+                else:
+                    logger.info("Recycle image(selected): ", ean, field, "->", image)
+
+
+def check_image_in_pack(image, lang, pack_images):
+    """
+    Check if image has been already selected
+
+    :param image: imgid
+    :param lang: en/fr/...
+    :param pack_images: image ids [packaging_en, front_en]
+    :return:
+    """
+    if "packaging_{}".format(lang) in pack_images:
+        current_id = pack_images.get("packaging_{}".format(lang), {}).get("imgid")
+        if current_id == image:
+            return True
+
+    return False
+
+
+def unselect_image(barcode, field_name, username, password):
+    """
+    Unselect image for product
+
+    :param barcode: ean
+    :param field_name: packaging_en
+    :param username:
+    :param password:
+    :return:
+    """
+
+    url = "https://world.openfoodfacts.org/cgi/product_image_unselect.pl"
+    data = {
+        "code": barcode,
+        "id": field_name,
+        "user_id": username,
+        "password": password,
+    }
+    try:
+        r = requests.post(url, data=data)
+        if r.ok:
+            return True
+    except Exception:
+        logger.warn(
+            "Exception in unselect_image: barcode - {}, id - {}".format(
+                barcode, field_name
+            )
+        )
+
+    return False
+
+
+def reselect_image(barcode, field_name, img_id, username, password):
+    """
+    Select image for product
+
+    :param barcode: ean
+    :param field_name: packaging_en
+    :param img_id: imgid=[0, 1, 2]
+    :param username:
+    :param password:
+    :return:
+    """
+
+    if img_id:
+        url = "https://world.openfoodfacts.org/cgi/product_image_crop.pl"
+        data = {
+            "code": barcode,
+            "imgid": img_id,
+            "id": field_name,
+            "user_id": username,
+            "password": password,
+        }
+        try:
+            r = requests.post(url, data=data)
+            if r.ok:
+                return True
```

Comment:
This, together with the final `return False`, could be rewritten as:

```suggestion
            return r.ok
```

Reply:
In general yes, if it helps